Lib/pdb.py

```diff
@@ -1719,10 +1753,24 @@ def _help_message_from_doc(self, doc):
             elif i < usage_end:
                 prefix = "       "
             else:
+                if usage_only:
+                    break
                 prefix = ""
             formatted.append(indent + prefix + line)
         return "\n".join(formatted)
 
+    def _print_invalid_arg(self, arg):
+        """Return the usage string for a function."""
+
+        self.error(f"Invalid argument: {arg}")
+
+        # Yes it's a bit hacky. Get the caller name, get the method based on
+        # that name, and get the docstring from that method.
+        # This should NOT fail if the caller is a method of this class.
+        doc = inspect.getdoc(getattr(self, sys._getframe().f_back.f_code.co_name))
```

Comment:
You can avoid the `f_back` by passing a depth to `sys._getframe`:

```suggestion
        doc = inspect.getdoc(getattr(self, sys._getframe(1).f_code.co_name))
```

This can also be cleaned up a bit further with the knowledge that `co_consts[0]` is always the original docstring of a function:

```suggestion
        doc = sys._getframe(1).f_code.co_consts[0]
```

Your call if that feels a little too implementation-detail-y. I personally think it's fine (as long as it has a comment).

Reply:
I was aware that `co_consts[0]` is the docstring of the function, but I'm not confident relying on that. It feels to me that it's an implementation detail and could potentially change in the future. It's also not documented as far as I know.